source/slang/slang-ast-builder.h

```diff
@@ -435,17 +434,17 @@ class ASTBuilder : public RefObject
     Val* getSNormModifierVal();
     Val* getNoDiffModifierVal();
 
-    Type* getTupleType(List<Type*>& types);
+    TupleType* getTupleType(List<Type*>& types);
 
-    Type* getFuncType(List<Type*> parameters, Type* result);
+    FuncType* getFuncType(ArrayView<Type*> parameters, Type* result, Type* errorType = nullptr);
 
     TypeType* getTypeType(Type* type);
 
         /// Produce a witness that `T : T` for any type `T`
     TypeEqualityWitness* getTypeEqualityWitness(
         Type* type);
 
-    SubtypeWitness* getDeclaredSubtypeWitness(
+    DeclaredSubtypeWitness* getDeclaredSubtypeWitness(
```

Comment:
I think I left this return type as `SubtypeWitness` because of the fact that if there is a constraint like `T : IFoo & IBar` we might want to decompose that ahead of time, by extracting the two sub-witnesses for `T : IFoo` and `T : IBar` and then re-composing them into a conjunction.

The approach I ended up using for canonicalization/simplification of subtype witnesses ends up not needing/wanting that step, but I was hoping that we could avoid downstream code accidentally depending on the exact class of AST node that gets returned by these various factory functions.

(All that said, I'm assuming the signature changed here because some other code needed the specificity...)